aioxmpp/ibb/service.py

```diff
@@ -0,0 +1,507 @@
+########################################################################
+# File name: service.py
+# This file is part of: aioxmpp
+#
+# LICENSE
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program.  If not, see
+# <http://www.gnu.org/licenses/>.
+#
+########################################################################
+import asyncio
+import base64
+import contextlib
+import os
+
+import aioxmpp
+import aioxmpp.errors as errors
+import aioxmpp.callbacks
+import aioxmpp.service as service
+
+from . import xso as ibb_xso
+
+
+MAX_BLOCK_SIZE = (1<<16) - 1
+
+
+def _generate_session_id():
+    # XXX: should we use secrets.token
+    return base64.urlsafe_b64encode(os.urandom(12)).decode("us-ascii")
```

Comment:
As discussed, we probably want to refactor the stanza ID generator function into utils and use that instead.